pkgs/development/tools/misc/elfutils/default.nix

```diff
@@ -1,34 +1,40 @@
-{ lib, stdenv, fetchurl, m4, zlib, bzip2, bison, flex, gettext, xz, setupDebugInfoDirs, argp-standalone }:
+{ lib, stdenv, fetchurl, pkg-config, autoreconfHook, musl-fts, obstack, m4, zlib, bzip2, bison, flex, gettext, xz, setupDebugInfoDirs, argp-standalone }:
 
 # TODO: Look at the hardcoded paths to kernel, modules etc.
 stdenv.mkDerivation rec {
   pname = "elfutils";
-  version = "0.180";
+  version = "0.182";
 
   src = fetchurl {
     url = "https://sourceware.org/elfutils/ftp/${version}/${pname}-${version}.tar.bz2";
-    sha256 = "17an1f67bfzxin482nbcxdl5qvywm27i9kypjyx8ilarbkivc9xq";
+    sha256 = "7MQGkU7fM18Lf8CE6+bEYMTW1Rdb/dZojBx42RRriFg=";
   };
 
-  patches = [ ./debug-info-from-env.patch ];
+  patches = [ ./debug-info-from-env.patch ]
+    ++ lib.optional stdenv.hostPlatform.isMusl ./musl-compat.patch;
```

Comment:
Can we apply this patch all the time?

Reply:
We could, though the patch adds extra warnings about null pointer derefs in both versions. This is just cribbed from alpine and nearly unmodified, but didn't want to introduce a warning suppress where it's not needed. I could try and work the patch and see if I can make the compiler happy, but it's an unfamilar codebase and wasn't feeling all the way up to it.

Open to any option, applying and suppressing, keeping them separate with a note to come back, or trying to tweak the patch.

Reply:
The problem with applying patches conditional is that they will bit rot because no one will test them on the next update.

Reply:
OK, so I switched around the patch stuff to just pull from alpine's git repo, and only conditionally-applying the one making up for musl's lack of error.h, which is the one that causes the null-pointer warning (again, not sure why). also applies a patch that fixes the aarch64 musl build. PTAL.

Reply:
ah, right, well, this patch now makes the aarch64 build succeed if pkgsMusl.coreutils succeeds, which it does not right now due to a test failure i can't quite pin down.